src/Interpreters/JoinToSubqueryTransformVisitor.cpp

```diff
@@ -73,16 +74,31 @@ class ExtractAsterisksMatcher
             }
         }
 
-        void addTableColumns(const String & table_name)
+        struct AlwaysAddColumnPredicate
+        {
+            bool operator()(const String & column_name [[maybe_unused]])
+            {
+                return true;
+            }
+        };
+
+        template <typename Predicate = AlwaysAddColumnPredicate>
+        void addTableColumns(const String & table_name, Predicate && should_add_column_predicate = Predicate())
```

Comment:
Why not use `std::function<bool(const String &)> = {}`?

Reply:
Updated.